backend/aws-parameterstore.go

```diff
@@ -0,0 +1,123 @@
+package backend
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ssm"
+)
+
+// AWSParameterStore implements Backend for AWS Parameter Store service.
+type AWSParameterStore struct {
+	client   *ssm.SSM
+	kmsKeyID string
+}
+
+// NewAWSParameterStore returns a new AWSParameterStore with an initialized AWS SSM client.
+func NewAWSParameterStore(ctx context.Context, p *Profile) (*AWSParameterStore, error) {
+	sess := session.New(&aws.Config{
+		Region:     aws.String(p.Location),
+		MaxRetries: aws.Int(2),
```

Comment:
Should we consider this a parameter in the backend configuration, next to a default value??